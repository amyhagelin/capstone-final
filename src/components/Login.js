import React from "react";
import {connect} from 'react-redux';
import { Link } from 'react-router-dom'

import { login } from "../actions/auth";

export class Login extends React.Component {

    auth(event) {
        event.preventDefault();
        console.log(this.props);
        const usernameValue = this.usernameInput.value;
        const passwordValue = this.passwordInput.value;
        this.props.dispatch(login(usernameValue, passwordValue));
    }

// add password like usernameValue DONE
// do the same with signup, model will be the same as User ERROR
// change the post route from /login to /signup in server.js ERROR
// think about features, what models I need for features, 
// what api request I need (GET events, POST events)
// create the schemas for the other features

    render() {
        return (
            <div>
                <header><Link to="/">Home</Link></header>
                <section className="min-height abstract-background">
                    {/*{ this.props.state.ui.alert && this.props.state.ui.alert.location === 'login' && <Alert /> }*/}
                    { this.props.alert && this.props.alert.title }
                    <form className="nav-padding">
                        <input placeholder="YOUR USERNAME" type="text" ref={ input => this.usernameInput = input }/>
                        <br/>
                        <input placeholder="YOUR PASSWORD" type="text" ref={ input => this.passwordInput = input }/>
                        <br/>
                        <button className="button" onClick={e => this.auth(e)} >
                         Log In                       
                        </button>
                    </form>
                    {/*{ isErrorForLoginPage && <Alert errorMessage={} /> }*/}
                    <div className="alert">
                        { this.props.alert && this.props.alert.title }
                    </div>
                </section>
            </div>
        );
    }

};



const mapStateToProps = state => ({
    alert: state.ui.alert
});


export default connect(mapStateToProps)(Login);