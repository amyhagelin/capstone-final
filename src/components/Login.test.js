import React from 'react';
import { shallow, mount } from 'enzyme';

import { Login } from './Login';

xdescribe ('<Login />', () => {
    it('Renders without crashing', () => {
        shallow(<Login />);
    });

    it('Renders with proper data', () => {
        const wrapper = shallow(<Login 
            dispatch={ () => {} }  // how do I do this?
            eventLog={ [] }
        />);

        expect(wrapper.find('h2').length).toEqual(1)
    });


});