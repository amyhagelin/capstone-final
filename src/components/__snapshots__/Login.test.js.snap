// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders correctly 1`] = `
<div>
  <header>
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </header>
  <section
    className="min-height abstract-background"
  >
    <form
      className="nav-padding"
    >
      <h3>
        To enter demo mode use
        <br />
        Username: demomode / Password: 123
      </h3>
      <input
        placeholder="Your Username"
        type="text"
      />
      <br />
      <input
        placeholder="Your Password"
        type="password"
      />
      <br />
      <button
        className="button"
        onClick={[Function]}
      >
        Log In
      </button>
    </form>
    <div
      className="alert"
    />
  </section>
</div>
`;
